modules/kubernetes-addons/crossplane/main.tf

```diff
@@ -14,5 +14,46 @@ module "helm_addon" {
   helm_config       = local.helm_config
   irsa_config       = null
 
-  depends_on = [kubernetes_namespace_v1.irsa]
+  depends_on = [kubernetes_namespace_v1.crossplane]
+}
+
+resource "kubectl_manifest" "aws_provider" {
+  for_each  = toset(data.kubectl_path_documents.aws_provider.documents)
+  yaml_body = each.value
+
+  depends_on = [module.helm_addon]
+}
+
+module "aws_provider_irsa" {
+  source                            = "../../../modules/irsa"
+  eks_cluster_id                    = var.eks_cluster_id
+  create_kubernetes_namespace       = false
+  create_kubernetes_service_account = false
+  kubernetes_namespace              = local.namespace
+  kubernetes_service_account        = "provider-aws-*"
+  irsa_iam_policies                 = concat([aws_iam_policy.aws_provider.arn], var.crossplane_provider_aws.additional_irsa_policies)
+  tags                              = var.tags
+
+  depends_on = [kubectl_manifest.aws_provider]
+}
+
+resource "aws_iam_policy" "aws_provider" {
+  description = "Crossplane AWS Provider IAM policy"
+  name        = "${var.eks_cluster_id}-aws-provider-irsa"
+  policy      = data.aws_iam_policy_document.s3_policy.json
+  tags        = var.tags
+}
+
+# Wait for the AWS Provider CRDs to be fully created before initiating aws_provider_config deployment
+resource "time_sleep" "wait_30_seconds" {
```

Comment:
Isn't `depends_on = [kubectl_manifest.aws_provider]` (in line 58) sufficient here?as it should deploy the CRDs manifests and then continue here

Reply:
`kubectl_apply` doesn't wait for all the CRDs to be applied here. It comes back with successfully completed however the CRDs are still being deployed.

I have hit few issues during the deployment so this logic came in handy to avoid the issue.

Reply:
Thanks! Let me try this one now to see if we can replace the time resource

Reply:
But the docs says  `finalized to complete for deleted objects`. We are creating the objects here

```[wait](https://registry.terraform.io/providers/gavinbunney/kubectl/latest/docs/resources/kubectl_manifest#wait) - Optional. Set this flag to wait or not for finalized to complete for deleted objects. Default false.```